### PR TITLE
Implement basic rate limit exception

### PR DESCRIPTION
**Pull request by Discordius**
┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1206291881933182) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/server/rateLimitUtils.ts [hunk 1]
```diff
@@ -89,6 +90,13 @@ async function getCommentsInTimeframe(userId: string, maxTimeframe: number) {
  * On forums other than the EA Forum, the post author is always exempt on their own posts.
  */
 async function shouldIgnoreCommentRateLimit(user: DbUser, postId: string|null, context: ResolverContext): Promise<boolean> {
+  const isRateLimitExempt = await ModeratorActions.findOne({
+    userId: user._id,
+    type: EXEMPT_FROM_RATE_LIMITS,
+    endedAt: { $gt: new Date()}
+  })
+  if (isRateLimitExempt) return true
```
Comment (b0b3rt) [inline packages/lesswrong/server/rateLimitUtils.ts:98]: I'd put this after the non-async checks, to save some pointless round trips

File: packages/lesswrong/lib/rateLimits/utils.ts [hunk 2]
```diff
@@ -30,7 +32,14 @@ export function getMaxAutoLimitHours(rateLimits?: Array<AutoRateLimit>) {
...
 
-export function shouldIgnorePostRateLimit(user: DbUser) {
+export async function shouldIgnorePostRateLimit(user: DbUser) {
+  const isRateLimitExempt = await ModeratorActions.findOne({
+    userId: user._id,
+    type: EXEMPT_FROM_RATE_LIMITS,
+    endedAt: { $gt: new Date()}
+  })
+  if (isRateLimitExempt) return true
+
```
Comment (b0b3rt) [inline packages/lesswrong/lib/rateLimits/utils.ts:42]: I'd put this after the non-async checks, to save some pointless round trips

File: packages/lesswrong/server/rateLimitUtils.ts [hunk 3]
```diff
@@ -89,6 +90,13 @@ async function getCommentsInTimeframe(userId: string, maxTimeframe: number) {
  * On forums other than the EA Forum, the post author is always exempt on their own posts.
  */
 async function shouldIgnoreCommentRateLimit(user: DbUser, postId: string|null, context: ResolverContext): Promise<boolean> {
+  const isRateLimitExempt = await ModeratorActions.findOne({
+    userId: user._id,
+    type: EXEMPT_FROM_RATE_LIMITS,
+    endedAt: { $gt: new Date()}
```
Comment (b0b3rt) [inline packages/lesswrong/server/rateLimitUtils.ts:96]: nit: missing space after `new Date()` to match the opening space

File: packages/lesswrong/lib/rateLimits/utils.ts [hunk 4]
```diff
@@ -30,7 +32,14 @@ export function getMaxAutoLimitHours(rateLimits?: Array<AutoRateLimit>) {
   }))
 }
 
-export function shouldIgnorePostRateLimit(user: DbUser) {
+export async function shouldIgnorePostRateLimit(user: DbUser) {
+  const isRateLimitExempt = await ModeratorActions.findOne({
+    userId: user._id,
+    type: EXEMPT_FROM_RATE_LIMITS,
+    endedAt: { $gt: new Date()}
```
Comment (b0b3rt) [inline packages/lesswrong/lib/rateLimits/utils.ts:39]: nit: missing space after `new Date()` to match the opening space

File: packages/lesswrong/server/rateLimitUtils.ts (hunk 1 above)
Comment (Discordius) [reply packages/lesswrong/server/rateLimitUtils.ts:98]: Oh, very reasonable

**Review by b0b3rt — APPROVED** [review]

**Pull request merged**